docs/contributing/3.0-development/plugin-layout.rst

```diff
@@ -0,0 +1,86 @@
+Pulp Plugin Application Layout
+==============================
+
+This documentation is for Pulp Plugin developers. For Platform development,
+see the :doc:`app-layout`.
+
+The Pulp Platform does not manage any content itself. This functionality is provided by
+its plugins, which use the Pulp Platform Plugin API to manage specific types of content,
+like RPM Packages or Puppet Modules. To do this, the Pulp Platform extends the Django
+Web Framework and the Django REST Framework to provide a set of base classes that can be
```

Comment:
None of the DRF classes are abstract (you can use ModelSerializer directly, you can use HyperlinkedRelatedField directly, etc), some of the Django base classes are not abstract.